ls1intum/Artemis
# `Notifications`: Add new notification, email, settings for successful file submissions

### Checklist
#### General
#### Server
 calls.
#### Client

### Motivation and Context
This PR is part of the initiative to resolve (as many as possible) open GitHub issues.
I noticed this stale issue #1089 and decided to resolve it. I talked with @krusche and he confirmed that I should add this new notification type.

### Description
This PR adds a new notification type, email, and settings for successful file submissions.

### Steps for Testing
Prerequisites:
- 1 User
- 1 File Upload Exercise

1) Use TS1 (only TS with email support) with your personal tum account (make sure you have access to the email address (can be checked in account settings) (otherwise you will not be able to receive emails)
2) Open your notification settings and make sure the new (file submit successful) setting is deactivated by default
3) Activate the new setting for webapp and email
4) Find a file upload exercise and upload/submit a file (e.g. png)
5) Make sure you can receive a webapp notification (in Artemis) and an email.
6) Check if the notification/email are correct (e.g. links work, typos etc.)

### Review Progress

#### Code Review
#### Manual Tests

### Test Coverage
I adjusted the touched testfiles
SingleUserNotificationService ~100%
SingleUserNotificationFactory > 90%
NotificationSettingsService > 90%
notification-settings.service = 100%

![image](https://user-images.githubusercontent.com/65814168/142697172-e6951603-4208-4f0c-b864-e3bf54ea6606.png)


### Screenshots
The new setting 
![image](https://user-images.githubusercontent.com/65814168/142694289-d44f667c-a110-4dc2-867b-8fce1c2cc1b2.png)

The new (sidebar) notification
![image](https://user-images.githubusercontent.com/65814168/142694409-8f0feb88-731d-496e-857a-971d4b682b3e.png)

New Email
![image](https://user-images.githubusercontent.com/65814168/142694506-b66f0972-3e29-4b30-819f-9b0668d6b04e.png)
